### PR TITLE
transport/grpchttp2: add doc to methods and values

### PR DESCRIPTION
**Pull request by printchard**
Add documentation to all exported methods and values.

RELEASE NOTES: none

### CODE REVIEW

File: internal/transport/grpchttp2/framer.go [hunk 1]
```diff
@@ -121,10 +124,12 @@ type DataFrame struct {
 	Data []byte
 }
 
+// Header returns the 9 byte header for this frame.
```
Comment (easwars) [inline internal/transport/grpchttp2/framer.go:127]: Nit: s/`9 byte header`/`9 byte HTTP/2 header`/

File: internal/transport/grpchttp2/framer.go [hunk 2]
```diff
@@ -141,10 +146,12 @@ type HeadersFrame struct {
 	HdrBlock []byte
 }
 
+// Header returns the 9 byte header for this frame.
```
Comment (easwars) [inline internal/transport/grpchttp2/framer.go:149]: Same as the other comment. Here and elsewhere in this PR.

Comment (printchard) [reply internal/transport/grpchttp2/framer.go]: Done

File: internal/transport/grpchttp2/framer.go (hunk 1 above)
Comment (printchard) [reply internal/transport/grpchttp2/framer.go]: Done

**Review by easwars — APPROVED** [review]

**Comment by easwars** [comment]
Moving to @arvindbr8 for second set of eyes.

File: internal/transport/grpchttp2/errors.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ import "fmt"
 // [GOAWAY]: https://httpwg.org/specs/rfc7540.html#GOAWAY
 type ErrCode uint32
 
+// Error Codes defined by the HTTP/2 Spec.
```
Comment (arvindbr8) [inline internal/transport/grpchttp2/errors.go:33]: I'm not entirely sure how the linter works, but do you know if it expects you to add a docstring for each individual exported variable?

File: internal/transport/grpchttp2/framer.go [hunk 4]
```diff
@@ -27,6 +27,7 @@ import "golang.org/x/net/http2/hpack"
 // [Frame Type]: https://httpwg.org/specs/rfc7540.html#FrameType
 type FrameType uint8
 
+// Frame types defined in the HTTP/2 Spec.
```
Comment (arvindbr8) [inline internal/transport/grpchttp2/framer.go:30]: same here

**Review by arvindbr8 — APPROVED** [review]
LGTM. I have a question about how the linter works

**Pull request merged**